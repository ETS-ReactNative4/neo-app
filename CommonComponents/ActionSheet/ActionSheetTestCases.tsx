import React, { useRef, useState, Fragment } from "react";
import { View, Text } from "react-native";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import ActionSheet from "./ActionSheet";
import { StyleSheet } from "react-native";
import { CONSTANT_white1 } from "../../constants/colorPallete";
import Interactable from "react-native-interactable";
// @ts-ignore
import { responsiveHeight } from "react-native-responsive-dimensions";
import SimpleButton from "../SimpleButton/SimpleButton";

const ActionSheetHandler = () => {
  const actionSheetRef = useRef(null);
  const actionSheetStartingPosition = responsiveHeight(75);
  const [snappedIndex, setSnappedIndex] = useState(1);

  const openActionSheet = () => {
    // @ts-ignore
    actionSheetRef.current && actionSheetRef.current.snapTo({ index: 1 });
    return null;
  };

  const sheetSnapped = (snappedInfo: Interactable.ISnapEvent) => {
    setSnappedIndex(snappedInfo.nativeEvent.index);
  };

  return (
    <Fragment>
      <View style={styles.actionSheetHandler}>
        <SimpleButton
          text={"Open Panel"}
          textColor={"white"}
          action={openActionSheet}
        />
        <Text style={styles.infoText}>{`Action sheet is
  ${
    snappedIndex === 2
      ? "Hidden"
      : snappedIndex === 1
      ? "Visible"
      : "Full Screen"
  }`}</Text>
      </View>
      <ActionSheet
        onSnap={sheetSnapped}
        interactableRef={actionSheetRef}
        panelStartingPosition={actionSheetStartingPosition}
      >
        <Text style={styles.infoText}>This is the Action sheet!</Text>
      </ActionSheet>
    </Fragment>
  );
};

const styles = StyleSheet.create({
  actionSheetHandler: {
    flex: 1,
    backgroundColor: CONSTANT_white1,
    alignItems: "center",
    justifyContent: "center"
  },
  infoText: {
    textAlign: "center",
    margin: 16,
    fontWeight: "bold"
  }
});

const ActionSheetTestCases: ITestCase[] = [
  {
    title: "Action Sheet",
    Component: <ActionSheet />
  },
  {
    title: "Action Sheet Props",
    Component: <ActionSheetHandler />
  }
];

export default ActionSheetTestCases;
