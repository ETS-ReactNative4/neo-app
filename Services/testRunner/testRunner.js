import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
