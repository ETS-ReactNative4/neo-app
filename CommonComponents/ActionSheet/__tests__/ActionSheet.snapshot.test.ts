import ActionSheetTestCases from "../ActionSheetTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

ActionSheetTestCases.map(testRunner);
