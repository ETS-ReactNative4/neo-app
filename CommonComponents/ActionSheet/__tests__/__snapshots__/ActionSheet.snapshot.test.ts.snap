// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Sheet 1`] = `
Array [
  <View
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "black",
        "height": 1334,
        "opacity": 0.29999999999999993,
        "position": "absolute",
        "width": 750,
      }
    }
  />,
  <InteractableView
    animatedNativeDriver={true}
    dragToss={0.1}
    initialPosition={
      Object {
        "y": 333.5,
      }
    }
    onAnimatedEvent={[Function]}
    onSnap={[Function]}
    reportOnAnimatedEvents={true}
    snapPoints={
      Array [
        Object {
          "y": 0,
        },
        Object {
          "y": 333.5,
        },
        Object {
          "y": 1434,
        },
      ]
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 30,
        "height": 2668,
        "overflow": "hidden",
        "position": "absolute",
        "width": 750,
      }
    }
    verticalOnly={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 30,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(221, 221, 221, 1)",
            "borderRadius": 3,
            "height": 8,
            "marginTop": 16,
            "width": 40,
          }
        }
      />
    </View>
  </InteractableView>,
]
`;

exports[`Action Sheet Props 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(247, 247, 247)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      onPress={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,198,132,1)",
            "borderRadius": 4,
            "height": 40,
            "width": 160,
          },
          Object {},
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 17,
                "lineHeight": 17,
                "marginTop": 2,
              },
              Object {
                "color": "white",
              },
              Object {},
            ]
          }
        >
          Open Panel
        </Text>
      </View>
    </TouchableHighlight>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "margin": 16,
          "textAlign": "center",
        }
      }
    >
      Action sheet is
  Hidden
    </Text>
  </View>,
  <View
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "black",
        "height": 1334,
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    }
  />,
  <InteractableView
    animatedNativeDriver={true}
    dragToss={0.1}
    initialPosition={
      Object {
        "y": 1334,
      }
    }
    onAnimatedEvent={[Function]}
    onSnap={[Function]}
    reportOnAnimatedEvents={true}
    snapPoints={
      Array [
        Object {
          "y": 0,
        },
        Object {
          "y": 1000.5,
        },
        Object {
          "y": 1434,
        },
      ]
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 30,
        "height": 2668,
        "overflow": "hidden",
        "position": "absolute",
        "width": 750,
      }
    }
    verticalOnly={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 30,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(221, 221, 221, 1)",
            "borderRadius": 3,
            "height": 8,
            "marginTop": 16,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "margin": 16,
            "textAlign": "center",
          }
        }
      >
        This is the Action sheet!
      </Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "margin": 16,
            "textAlign": "center",
          }
        }
      >
        Action sheet is
  Hidden
      </Text>
    </View>
  </InteractableView>,
]
`;

exports[`Action Sheet with Text Input 1`] = `
Array [
  <View
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "black",
        "height": 1334,
        "opacity": 0.29999999999999993,
        "position": "absolute",
        "width": 750,
      }
    }
  />,
  <InteractableView
    animatedNativeDriver={true}
    dragToss={0.1}
    initialPosition={
      Object {
        "y": 333.5,
      }
    }
    onAnimatedEvent={[Function]}
    onSnap={[Function]}
    reportOnAnimatedEvents={true}
    snapPoints={
      Array [
        Object {
          "y": 0,
        },
        Object {
          "y": 333.5,
        },
        Object {
          "y": 1434,
        },
      ]
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 30,
        "height": 2668,
        "overflow": "hidden",
        "position": "absolute",
        "width": 750,
      }
    }
    verticalOnly={true}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 30,
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgba(221, 221, 221, 1)",
            "borderRadius": 3,
            "height": 8,
            "marginTop": 16,
            "width": 40,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        placeholder="Tap here to show keyboard!"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "marginTop": 16,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </InteractableView>,
]
`;
